### ISSUE TITLE
Incorrect Package Name

### ISSUE THREAD

**Issue by ikuser** [spring-petclinic/spring-petclinic-graphql#5]
package org.springframework.samples.petclinic.service;
Change this to - package org.springframework.samples.petclinic.repository;

Please add me to this project contributor. I would like to contribute.

**Comment by arey** [comment]
Hi @ikuser you could contribute with a pull request :)

**Comment by ikuser** [comment]
Hi Antonie,

I need some help in setting my work-space.


There are some error I am seeing, due to apollo-client 2.0+ version update and other git code versions.

Current Git source is using 1.4 (apollo client), and upgrading any package ( Typescript, React etc ) is breaking other code and compilation errors.


Do you please think, can I call you and video share to help me setup this new GraphQL stack - frontend code.

Also, I need some info on how to make new pull request for me to contribute.


I can be reached at - 848 248 1634 or please give me your number/time I will call you then please.


Thank you

Imran.

________________________________

**Comment by arey** [comment]
Hi Imran,

I don't understand what problems you are facing: apollo client upgrade? setting the project workspace in your IDE?
Do you have any Skype account to call you?
For my part, I don't know GraphQL. @nilshartmann is the main contributor of this project.
@nilshartmann do you plan to upgrade appolo client version? 
The GitHub pull request guide is available here: https://help.github.com/articles/about-pull-requests/

Regards

Antoine

**Comment by ikuser** [comment]
Hi Antoine,

First I want to thank you all. What you guys have shared and knowledge everybody get from your

sharing the code repo is really good.


My Skype id is : ikusa786 or add via my email (ik_usa@hotmail.com).

Normally I never use Skype, but let me know the day/time and I will be available ( preferably evening

EST after 5:00 pm, regular work hours).


Info on below question....

Also attaching compilation error file:


--- I don't understand what problems you are facing: apollo client upgrade? setting the project workspace in your IDE?

      ----- apollo client upgrade? Yes, if I upgrade apollo client ( to 2.0 +) and also React JS ( types and other version from 15-16). Those types break the code, and there are some changes in Apollo client connection mechanism that needs to be modified as well.

     ----- setting the project workspace in your IDE?   ---- Yes, I think i will need some info on setting work-space for GraphGl server part. Though for last 2 days i reverted (NodeJS packages )to version what is currently in Git, but i think that GraphGL server configuration is incorrect, which is giving me compilation error


Here are some pieces I have identified, This is based on comparison between,

https://github.com/spring-petclinic/spring-petclinic-reactjs
https://github.com/spring-petclinic/spring-petclinic-graphql

I think graphql - is little more proper updated version, of the code.
Basically some logic or those profiling during execution, test case Abstract class execution, DB (init, execution, configuration options) etc is little bit missing(or little different) in regular reactjs project (which uses REST).

The part is I am also new to GraphGL. And I clearly see the benefit of using it, but due to lack
of knowledge and setup issue I have to switch to REST and migrate in slower pace to GraphGL.
Also I need to understand the setup in Cloud for GraphGL part.

Please help me in GraphGL local workspace setup (if u can). I would love to see what I can add from my side.

Thank you
Imran.



________________________________

**Comment by nilshartmann** [comment]
Hi Imran,

sorry for jumping in so late.

It is very important that you use the correct versions of each javascript module, that is defined in the package.json file, esp the TypeScript, React and Apollo versions must fit together. There have been some major changes in newer versions of this tools, Apollo 2 for example is quite different from Apollo 1.

The best way to make sure that you're using the correct version is, to use [yarn](https://yarnpkg.com), as written in the README file in the root folder of this repository.

The compilation error you pasted  is not from Java (Server-part) but from TypeScript (that is used in the frontend). That might be due to wrong javascript packages that can be avoided by using yarn instead of npm.

BTW: If you're new to GraphQL and just want to learn it, this project might not be the best place to start, because it's a quite heavy technology stack. The project is more appropriate for people already knowing grahql and want to see how to combine SpringBoot/GraphQL on the server and GraphQL/React/TypeScript on the client.

So if you want to learn GraphQL:

1. read the docs at http://graphql.org/learn/

2. choose an implementation for the serverside (Java, JavaScript, depending on your needs). Using the Spring Boot Starter for GraphQL might not be the best way to start, because it abstracts a lot of things (which is helpful, but not helpful for actually *learning* the server-side implementation). If you're using java, I would recommend to start with the "pure" java implementation https://github.com/graphql-java/graphql-java

3. choose an implementation for your client. If you're frontend is JavaScript (and esp if it's React-based) consider using the Apollo Client (https://www.apollographql.com/) that I'm using in this project. (If you're not familiar with TypeScript, start with it's JavaScript API to reduce complexity and the amount of things you have to learn) The apollo client is very well documented and a solid implementation for the client-side.

(4. If you want to try out GraphQL queries, you could use GitHubs GraphQL API: https://developer.github.com/v4/ and https://developer.github.com/v4/explorer/)

Due to lack of time I cannot upgrade this project currently to Apollo 2.0. Maybe I will upgrade in April or May...

Hope that helps,
Nils

**Comment by ikuser** [comment]
Thank you so much Nils.

Your information was very good, It helped me quite lot on it.

I really hope and am looking forward to contributing with you guys.


Thank you

Imran.

________________________________
